src/bit_stream_reader/es_reader.cpp

```diff
@@ -0,0 +1,1163 @@
+
+    obu_size_ = 0;
+    obu_byte_offset_ = curr_byte_offset_;
+    // Parser header
+    if (GetByte(curr_byte_offset_, &header_byte) == false) {
+        return false;
+    }
+    *obu_type = (header_byte >> 3) & 0x0F;
+    obu_extension_flag = (header_byte >> 2) & 0x01;
+    curr_byte_offset_ = (curr_byte_offset_ + 1) % BS_RING_SIZE;
+    obu_size_++;
+    if (obu_extension_flag) {
+        curr_byte_offset_ = (curr_byte_offset_ + 1) % BS_RING_SIZE;
+        obu_size_++;
+    }
+    // Parse size
+    int len;
+    uint32_t value = 0;
+    uint8_t data_byte;
+    for (len = 0; len < 8; ++len) {
+        if (GetByte(curr_byte_offset_ + len, &data_byte) == false) {
+            return false;
+        }
+        value |= (data_byte & 0x7F) << (len * 7);
+        if ((data_byte & 0x80) == 0) {
+            ++len;
+            break;
+        }
+    }
+    obu_size_ += len + value;
+    curr_byte_offset_ = (curr_byte_offset_ + len + value) % BS_RING_SIZE;
+
+    return true;
+}
+
+bool RocVideoESParser::CopyObuFromRing() {
+    if (obu_size_ > GetDataSizeInRB()) {
+        if (FetchBitStream() == 0) {
+            end_of_stream_ = true;
+            return false;
+        }
+        if (obu_size_ > GetDataSizeInRB()) {
+            return false;
+        }
+    }
+    if ((pic_data_size_ + obu_size_) > pic_data_.size()) {
+        pic_data_.resize(pic_data_.size() + obu_size_);
+    }
+    int obu_end_offset = (obu_byte_offset_ + obu_size_) % BS_RING_SIZE;
+    if (obu_end_offset >= obu_byte_offset_) {
+        memcpy(&pic_data_[pic_data_size_], &bs_ring_[obu_byte_offset_], obu_size_);
+    } else {
+        memcpy(&pic_data_[pic_data_size_], &bs_ring_[obu_byte_offset_], BS_RING_SIZE - obu_byte_offset_);
+        memcpy(&pic_data_[pic_data_size_ + BS_RING_SIZE - obu_byte_offset_], &bs_ring_[0], obu_end_offset);
+    }
+    pic_data_size_ += obu_size_;
+    SetReadPointer(obu_end_offset);
+    return true;
+}
+
+int RocVideoESParser::GetPicDataAv1(uint8_t **p_pic_data, int *pic_size) {
+    int obu_type;
+    pic_data_size_ = 0;
+
+    while (!end_of_stream_) {
+        if (!ReadObuHeaderAndSize(&obu_type)) {
+            break;
+        }
+        CopyObuFromRing();
+        if (obu_type == kObuTemporalDelimiter) {
+            num_td_obus_++;
+        if (num_td_obus_ > 1) {
+            break;
+        }
+        }
+    }
+
+    *p_pic_data = pic_data_.data();
+    *pic_size = pic_data_size_;
+    num_temp_units_++;
+    return 0;
+}
+
+bool RocVideoESParser::CheckIvfFileHeader(uint8_t *stream) {
+    static const char *IVF_SIGNATURE = "DKIF";
+    uint8_t *ptr = stream;
+
+    // bytes 0-3: signature
+    if (memcmp(IVF_SIGNATURE, ptr, 4) == 0) {
+        ptr += 4;
+        // bytes 4-5: version (should be 0). Little Endian.
+        int ivf_version = ptr[0] | (ptr[1] << 8);
+        if (ivf_version != 0) {
+            ERR("Stream file error: Incorrect IVF version (" + TOSTR(ivf_version) + "). Should be 0.");
+        }
+        ptr += 2;
```

Comment:
better to combine below 2 statements to ptr += 4 and add comment accordingly

Reply:
Done.